README.md
# DAIL2
DAIL simplified for simple minded dev. 

DAIL is a mod compilation for the game Grim Dawn.

DAIL source is provided for those who wish to contribute to the project, or make edits for private use.
Otherwise, DAIL is not to be altered, renamed, ripped (other than for seeing how functions are set up and linked), 
or distributed for monetary gain.

The mods in DAIL are there by permission, or the author has an open license. The authors of the mods retain
creative ownership of the content. See the Features_and_Versions.txt file in the Dev_Info folder for a complete
listing of the mods and their authors.

A lot of time has been invested in making these mods work together by Davood, Kathanious, and AlienFromBeyond.
And a lot of time has been invested into making the individual mods by their authors. Please be respectful
with the content.


_Dev_info/changelog.txt
Edit [12/4/2016 19:30] K

****************************************************************
****************************************************************


To Do:
********

Weight Reduction 2nd Phase
	Header all files for mod identification
	Add GQ C & D to loottables and test
	
Update - DGA 0.5 
Update - Cornucopia 0.3.3 (Will include 1.0.0.6 new items)
Fix - Relic weapon reqs?
Update - Skills for staff/spear?

********
	
****************************************************************
****************************************************************

[60.7 12/5/2016]
Attempt 3 - Loot table alignment

Change - Cenobite
	Cleaned mastery UI icons

Change - Tag/UI conversion completed

Update - Necromancer (Enumeration 07)
	Pet folder from newest source

[60.7 12/4/2016]
Change - Begin DAIL second weight reduction pass
	Renamed playerclassN -> playerclassX (In preperation for expansion)
		playerclass07 -> playerclassnecro
		playerclass08 -> playerclassranger
		playerclass10 -> playerclassfangshi
		playerclass14 -> playerclassarnecro
		playerclass28 -> playerclassceno

Removed - All loot tables and chest tables

Change - Reduced B and X Mode Density
	Set Boss&Quest Proxies
		spawnMin,1,
		spawnMax,1,
		championMin,2,
		championMax,4,

Added - SD (S - Diminished) Mode
	Set Boss&Quest Proxies
		spawnMin,1,
		spawnMax,1,
		championMin,1,
		championMax,2,
		
[60.6 12/1/2016]
Added - Additional Davood merchandise - AlienFromBeyond 

[60.6 11/29/2016]
Update - Zenith 0.5B

Update - Omega 0.71

Update - Lost Treasures 0.25

Fixed - GQ & Ranger anm conflicts

[60.6 11/28/2016]
Post 50 mastery investment inquiries - AlienFromBeyond
	Grim Quest masteries
	Mystic Warrior
	
[60.6 11/27/2016]
Inquiry - Ranger Rapid Fire
	Animation conflict with GQ 

Fixed - Missing Merc combo names

[60.6 11/26/2016]
Remove - itemrecycler
	Causing crafting lock-up
	
Fixed - Missing TK training bar image

[60.5 11/24/2016]
Change - (X) and (SX)
	Lowered mob scaling

Change - Reverted Grimmest 1.0 to previous from XADAIL back-up

Change - dumped loot tables again - merged in listed order
	Merged - Cornucopia 
	Merged - GD Eternal
	Merged - Grim Quest
	Merged - Itemrecycler
	Merged - LCR

[60.4 11/23/2016]
Change - Loot tables - minimize mod density
	Archived current - 
	Merged - Cornucopia 
	Merged - GD Eternal
	Merged - Grim Quest
	Merged - Itemrecycler
		
Update - Grim Quest 1.4.4

Change - Occultist
	Increased allowable pets - pet lvl 1-5, 2 pet 5-10, 3 pet 11-15, 4 pet 16-26

[60.4 11/22/2016]
Inquiry - DGA Factions
	User14 - Merchants - Gained with killing enemies (User 15)
	Default factions DC (Survivors) and Aetherial on player Faction tab
		Factions not used in DGA - No DC merchant

Fixed - Missing GQ controller data

Fixed - Missing Cornucopia components

Update - Grimmest [1.0]
	Base for (S) amd (SX) density
	
[60.4 11/20/2016]
Update - Zenith 0.5

[60.3 11/20/2016]
Fixed - Missing DoH controllers

[60.2 11/20/2016]
Update - Omega 0.7

Fixed - Adjustment dbr for (B) and (S) modes

[60.2 11/18/2016]
Fixed - Malepc01.dbr class28 typo

[60.1 11/18/2016]
Update - Dawn of Heroes 3.9

Fixed - Missing Grover Keeper assets

Fixed - Incorrect mob challenge in B and S

[60 11/17/2016]
Update - Dawn of Heroes 3.8

[60 11/15/2016]
Update - Dawn of Heroes 3.7

Update - Lost Treasure 0.24

Update - Omega 0.67

[60 11/12/2016]
Rotate - Illusionist#15 -> Mystic Warrior

Rotate - Engineer#27 -> Mercenary

Update - Omege 0.65

Fixed - Arcanist - Aether Ray missing Aether damage (imagine that)

[59 11/10/2016]
Update - Lost Treasures 022.b

Update - Dawn of Heroes 3.6

[59 11/9/2016]
Purge - Working dir
	Pulled back-up 11/6/2016
	Reworked changes

[59 11/8/2016]
Change - all gameengine.dbr
	max pot 9999

Merged - Omega 0.64
	Notes : http://www.grimdawn.com/forums/showthread.php?t=37935&page=10
	Fix - all gameengine.dbr faction13

Inquiry - DGA file integrity
	No endless?

Inquiry - GDE file integrity
	No lower pantheon?
	
Update - NCFF 1.23
	
[58.3 11/6/2016]
	
Fixed - BoC Portal
	Wavespawn.lua removed from main.lua
	Moved record/level art to GDE Unique

Fixed - Ranger tag
	No tag in source - made something up

Fixed - Hunting enumeration
	23 -> 17 (Failure to correct after GQ update)
	
[58.2 11/5/2016]
Inquiry - Nullified mode specific folders
	Applied tags

Inquiry - Potion stack limit
	1000
	
Fixed - Rare crafting components not dropping
	autoloot entries in enemies dbrs

[58.1F 11/4/2016]
Update - GQ Hotfix
Miracle - Fix FK pet bug
	Added dHand animations records/creatures/enemies/anm/anm_skeleton_fast.dbr

[58.1 11/3/2016]
Update - NCFF 1.10

Fixed - Materia bug - Grimmer 1.0 tdyn loot weights

Reapplied - Hunting source (no javelin update in 1.4.3)

Reapplied - DBRs (materia bug)
	Corn materia/loottables
	GQ materia/loottables
	Grimmer enembies folder

[58 11/1/2016]
Update - GQ - 1.4.3

Added - Mastery background image eye candy

[58 10/30/2016]
Update - DAIL - Grimmer (B) - Grimmer 1.0 - Reverted update

[58 10/29/2016]
Merged - All play mode resources
	Simplified database
		More work on release day, less cross database errors

Fixed - Missing GQ resources
	GQ relics on faction tables

[58 10/26/2016]
Inquiry - FK Revenant pet
	Functions in test play

Fixed - Transmorg not selling illusions
	Replaced missing database/records/items/crafting/bluprints/transmogs folder

Inquiry - Cenobite Shield Bash (strikes on 2nd hit, spec 4th hit)
	Matches source
	Change - 1 charge level -> 3 (for 4th strike bonus)

Release - Experimental version for loot test

Fixed - Fangshi broken tag

Fixed - DemonHunter incorrect tag

Fixed - Witch Doctor missing tag on mastery tab 

[58 10/25/2016]
Inquiry - Loot dbrs
	Removed remaining Skelethon
	Unable to determine loot malfunction

[57 10/22/2016]
Update - Grim Quest v1.42
	GQ Mastery expansions

[57 10/21/2016]
Update - itemskills folder - jiaco
	Component/item granted skills to work with spears
	
[57 10/19/2016]
Inquiry - Frost Pet
	Works in test play

Experimental Release

DAIL - Working/source/scripts/LostTreasures/LostTreasures.lua
--Lost Treasures MOD scripts
--credit : okami29 at http://www.grimdawn.com/forums/showthread.php?t=42534
ltreasures = {}
print("LostTreasures loaded")

local ltguardDBRs = { "records/treasures/creatures/enemies/hero/guardian01_boar_h06.dbr", "records/treasures/creatures/enemies/hero/guardian02_chthoniandevourer_h05.dbr", "records/treasures/creatures/enemies/hero/guardian03_fleshhulk_h04.dbr","records/treasures/creatures/enemies/hero/guardian04_trollhalfcorrupted_h05.dbr",'records/treasures/creatures/enemies/hero/guardian06_bonerat_melee_h06.dbr','records/treasures/creatures/enemies/hero/guardian07_chthoniandreadguard_h05.dbr','records/treasures/creatures/enemies/hero/guardian08_chthonianfiend_h05.dbr','records/treasures/creatures/enemies/hero/guardian09_chthonianleech_h05.dbr','records/treasures/creatures/enemies/hero/guardian10_dermapteran_h05.dbr','records/treasures/creatures/enemies/hero/guardian11_dranghoul_warrior_h07.dbr','records/treasures/creatures/enemies/hero/guardian12_gazer_h06.dbr','records/treasures/creatures/enemies/hero/guardian13_ghost_h06.dbr','records/treasures/creatures/enemies/hero/guardian14_golemrock_h06.dbr','records/treasures/creatures/enemies/hero/guardian15_harpy_h06.dbr','records/treasures/creatures/enemies/hero/guardian16_human_h04.dbr','records/treasures/creatures/enemies/hero/guardian17_obsidiandefiler_h05.dbr','records/treasures/creatures/enemies/hero/guardian18_skeletalgolem_h05.dbr','records/treasures/creatures/enemies/hero/guardian19_spider_h06.dbr','records/treasures/creatures/enemies/hero/guardian20_waspgiant_h07.dbr' }
local ltspawnbonus = { 1,2,3,4,5,6,7,8,9,10,11,12,13,18,23,34,48,73,98 }
local ltnotificationkills = { 'tagT_onkill1','tagT_onkill2','tagT_onkill3','tagT_onkill4','tagT_onkill5','tagT_onkill6','tagT_onkill7','tagT_onkill8','tagT_onkill9','tagT_onkill10','tagT_onkill11','tagT_onkill12','tagT_onkill13','tagT_onkill14','tagT_onkill15','tagT_onkill16','tagT_onkill17','tagT_onkill18','tagT_onkill19','tagT_onkill20','tagT_onkill21','tagT_onkill22','tagT_onkill23','tagT_onkill24','tagT_onkill25' }
local totalGuards = table.getn(ltguardDBRs)
local spawnedltguardU = 0
local killedltguard = 0

local function CharacterCreateArgs()

	local averageLevel = Game.GetAveragePlayerLevel()
	local monsterLevel = 100
	monsterLevel = ((averageLevel+(averageLevel/50))+4)
	if monsterLevel < 85 then monsterLevel = 89 end
	return monsterLevel, nil
	
end

function ltreasures.summon(objectId)
	local newCreature = Entity.Create(objectId);
	local playerCoords = Game.GetLocalPlayer():GetCoords();
	newCreature:SetCoords(playerCoords);
end

function ltreasures.ononeshotremoved(objectId,place)
	print ("ltreasures.ononeshotdestroyed : ",objectId,"place : ",place,"Initial spawnedltguardU=",spawnedltguardU)	
	local coords = Entity.Get(objectId):GetCoords()
	local player = Game.GetLocalPlayer()
	print("ok table and coords =",coords)
	if Game.GetGameDifficulty() >= Game.Difficulty.Legendary then
		print("Difficulty is Ultimate")
		math.randomseed(Time.Now());
		local randomizer = math.random(1,totalGuards);
		print("randomizer",randomizer,"killedltguard=",killedltguard,"TimeNow=",Time.Now())
		local spawn = Character.Create (ltguardDBRs[randomizer], CharacterCreateArgs())
		spawn:SetCoords(coords)
		print("Coords x MAJOR",coords.xAxis.x)
	end
end


function ltreasures.ondieguardian(objectId)
		local coords = Entity.Get(objectId):GetCoords()
		print("ondieguardian coords=",coords)
		killedltguard = killedltguard+1
		UI.Notify(ltnotificationkills[killedltguard])
		--math.randomseed(Time.Now());
		local randomizer01 = math.random(1,100);
		local randomizerbonus = randomizer01 + ltspawnbonus[killedltguard]
		print("randomizer01 =",randomizer01,"randomizernonus=",randomizerbonus,"spawnedltguardU=",spawnedltguardU)
		if randomizerbonus > 98 and spawnedltguardU < 3 then
			local ultimateguardian = Entity.Create('records/treasures/creatures/enemies/boss&quest/treasures_guardian_ultimate.dbr')
			ultimateguardian:SetCoords(coords);
			--ltreasures.summon('records/treasures/creatures/enemies/boss&quest/treasures_guardian_ultimate.dbr')
			spawnedltguardU = spawnedltguardU +1
		print("spawnedltguardU=",spawnedltguardU)
		end
end

function ltreasures.ondrop(objectId)
	print("ondrop",objectId)
	local newCreature = Entity.Create('records/treasures/fx/skillsother/treasures_fxpak_lootbeam.dbr');
	local coords = Entity.Get(objectId):GetCoords()
	newCreature:SetCoords(coords);
	UI.Notify('tagT_ondrop');
end
function ltreasures.ondropdestroy(objectId)
	print("ondropdestroy",objectId)
	Entity.Get(objectId):Destroy()
	print("Destroying shoulder")
end
--------------START Call on destoyed One shot---------
function ltreasures.ononeshotremoved1(objectId)
ltreasures.ononeshotremoved(objectId,1)
end
function ltreasures.ononeshotremoved2(objectId)
ltreasures.ononeshotremoved(objectId,2)
end
function ltreasures.ononeshotremoved3(objectId)
ltreasures.ononeshotremoved(objectId,3)
end
function ltreasures.ononeshotremoved4(objectId)
ltreasures.ononeshotremoved(objectId,4)
end
function ltreasures.ononeshotremoved5(objectId)
ltreasures.ononeshotremoved(objectId,5)
end
function ltreasures.ononeshotremoved6(objectId)
ltreasures.ononeshotremoved(objectId,6)
end
function ltreasures.ononeshotremoved7(objectId)
ltreasures.ononeshotremoved(objectId,7)
end
function ltreasures.ononeshotremoved8(objectId)
ltreasures.ononeshotremoved(objectId,8)
end
function ltreasures.ononeshotremoved9(objectId)
ltreasures.ononeshotremoved(objectId,9)
end
function ltreasures.ononeshotremoved10(objectId)
ltreasures.ononeshotremoved(objectId,10)
end
function ltreasures.ononeshotremoved11(objectId)
ltreasures.ononeshotremoved(objectId,11)
end
function ltreasures.ononeshotremoved12(objectId)
ltreasures.ononeshotremoved(objectId,12)
end
function ltreasures.ononeshotremoved13(objectId)
ltreasures.ononeshotremoved(objectId,13)
end
function ltreasures.ononeshotremoved14(objectId)
ltreasures.ononeshotremoved(objectId,14)
end
function ltreasures.ononeshotremoved15(objectId)
ltreasures.ononeshotremoved(objectId,15)
end
function ltreasures.ononeshotremoved16(objectId)
ltreasures.ononeshotremoved(objectId,16)
end
function ltreasures.ononeshotremoved17(objectId)
ltreasures.ononeshotremoved(objectId,17)
end
function ltreasures.ononeshotremoved18(objectId)
ltreasures.ononeshotremoved(objectId,18)
end
function ltreasures.ononeshotremoved19(objectId)
ltreasures.ononeshotremoved(objectId,19)
end
function ltreasures.ononeshotremoved20(objectId)
ltreasures.ononeshotremoved(objectId,20)
end
function ltreasures.ononeshotremoved21(objectId)
ltreasures.ononeshotremoved(objectId,21)
end
function ltreasures.ononeshotremoved22(objectId)
ltreasures.ononeshotremoved(objectId,22)
end

--------------END Call on destoyed One shot---------